timescaledb/quick-start/java.md

```diff
@@ -0,0 +1,621 @@
+# Quick Start: Java and TimescaleDB
+
+## Goal
+
+This quick start guide is designed to get Java developers up and running with TimescaleDB as their database.
+In this tutorial, you'll learn how to:
+
+* [Connect Java to TimescaleDB](#new_database)
+* [Create a relational table](#create_table)
+* [Generate a hypertable](#generate_hypertable)
+* [Insert a batch of rows into TimescaleDB](#insert_rows)
+* [Execute queries on TimescaleDB](#execute_queries)
+
+## Pre-requisites
+
+To complete this tutorial, you will need a cursory knowledge of the Structured Query Language (SQL). 
+The tutorial will walk you through each SQL command, but it will be helpful if you've seen SQL before.
+
+To start, [install TimescaleDB][timescaledb-install]. Once your installation is complete,
+you can proceed to ingesting or creating sample data and finishing the tutorial.
+
+You will also need to install [Java Development Kit (JDK)][jdk] 
+and [PostgreSQL Java Database Connectivity (JDBC) Driver][pg-jdbc-driver] as well.
+All code is presented for Java 16 and above. 
+If you are working with older JDK versions, use legacy coding techniques.
+
+## Connect Java to TimescaleDB [](new-database)
+
+### Step 1: Create a new Java application
+
+For simplicity, we will use the application in a single file as an example. 
+You can use any of your favorite build tools, including `gradle` and `maven`.
+
+Create a separate directory and navigate to it. 
+In it, create a text file with name and extension `Main.java` and the following content:
+
+```java
+package com.timescale.java;
+
+public class Main {
+
+    public static void main(String... args) {
+        System.out.println("Hello, World!");
+    }
+}
+```
+
+From the command line in the current directory, try running the application with this command:
+
+```bash
+$ java Main.java
+```
+
+You should see the `Hello, World!` line output to your console. 
+In case of an error, refer to the documentation and check if the JDK was installed correctly.
+
+### Step 2: Import Postgres JDBC driver
+
+To work with the `PostgreSQL`, you need to import the appropriate `JDBC Driver`. 
+If you are using a dependency manager, include [PostgreSQL JDBC Driver as dependency][pg-jdbc-driver-dependency]. 
+In this case, download [jar artifact of JDBC Driver][pg-jdbc-driver-artefact] and place it next to the `Main.java` file.
+
+Now you can import the `JDBC Driver` into the Java application and display a list of available drivers for the check:
+
+```java
+package com.timescale.java;
+
+import java.sql.DriverManager;
+
+public class Main {
+
+    public static void main(String... args) {
+        DriverManager.drivers().forEach(System.out::println);
+    }
+}
+```
+
+Use this command to run all the following examples:
+
+```bash
+$ java -cp *.jar Main.java
```

Comment:
For some reason, this didn't work by default.

```bash
javac -cp postgresql-42.2.20.jar com/timescale/java/Main.java
com/timescale/java/Main.java:9: error: cannot find symbol
        DriverManager.drivers().forEach(System.out::println);
                     ^
  symbol:   method drivers()
  location: class DriverManager
1 error
```

Reply:
I'm trying a few attempts and I'm a bit lost on the java compiler flags.

```bash
java -cp *.jar com/timescale/java/Main                16:02:44
Error: Could not find or load main class com.timescale.java.Main
javac -cp *.jar com/timescale/java/Main.java          16:02:47
com/timescale/java/Main.java:9: error: cannot find symbol
        DriverManager.drivers().forEach(System.out::println);
                     ^
  symbol:   method drivers()
  location: class DriverManager
1 error
```

Reply:
I have [this jar](https://jdbc.postgresql.org/download/postgresql-42.2.20.jar) in my folder.

Reply:
I think this problem is related to the same thing: you tried to create folder structure com/timescaledb/java that should be skipped and replace just with folder contains 2 files: Main.java and postgres jdbc jar.